### PR TITLE
deprecate `targetObject` in `target_action_support`

### PR DESCRIPTION
**Pull request by bekzod**
not sure if this is still valid, and not sure about `id` and `until` values

### CODE REVIEW

**Review by rwjblue — COMMENTED** [review]
We should also deprecate in the `get` path (changing the `getTarget` method to use `_targetObject` directly).

**Comment by bekzod** [comment]
added depreciation to `get` too

File: packages/ember-runtime/lib/mixins/target_action_support.js [hunk 1]
```diff
@@ -9,7 +9,19 @@ import {
   Mixin,
   computed
 } from 'ember-metal';
-import { assert } from 'ember-debug';
+import { assert, deprecate } from 'ember-debug';
+
+function depricateTargetObject() {
```
Comment (runspired) [inline packages/ember-runtime/lib/mixins/target_action_support.js]: sp. deprecateTargetObject*

File: packages/ember-runtime/lib/mixins/target_action_support.js [hunk 2]
```diff
@@ -9,7 +9,19 @@ import {
...
+  deprecate(
+    'Using `targetObject` has been deprecated in TargetActionSupport Mixin',
+    false,
+    {
+      id: 'ember-runtime.mixins.target-action-support.targetObject',
+      until: '3.0.0'
```
Comment (runspired) [inline packages/ember-runtime/lib/mixins/target_action_support.js]: do we add `since` now for the svelte work, or is that information stored elsewhere?

Comment (bekzod) [inline packages/ember-runtime/lib/mixins/target_action_support.js]: cannot find any usage of `since: ` anywhere

Comment (rwjblue) [inline packages/ember-runtime/lib/mixins/target_action_support.js]: We add it to the features.json file

Comment (bekzod) [inline packages/ember-runtime/lib/mixins/target_action_support.js]: should I add it ? what version it should be ?

**Comment by krisselden** [comment]
We should deprecate the whole mixin.  This isn't used by anything in Ember, it is just exported.

**Comment by locks** [comment]
@bekzod hey! We discussed this deprecation, and we decided that `TargetActionSupport` should be deprecated entirely, like @krisselden suggested. Since it's private API that's not even used by Ember, the deprecation should be until the next LTS, which is 2.16.

**Comment by bekzod** [comment]
@locks hi, thanks for letting know, I think it is better to be handled by person who knows `TargetActionSupport` better, so closing this

**Pull request closed without merging**